src/search/search.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+
+import QuerySymbols from './querySymbols'
+import SearchResults from './result'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+import { Company } from '../types'
+import './search.css'
+
+interface SearchState {
+  readonly searchValue: string
+  readonly companies: Company[]
+}
+
+interface SearchProps {
+  readonly getInfo: (company: Company) => void
+  readonly companySymbols: Company[]
+}
+
+const initState: { searchValue: string; companies: Company[] } = {
+  searchValue: 'Apple Inc. (AAPL)',
+  companies: []
+}
+
+class Search extends React.PureComponent<SearchProps, SearchState> {
```

Comment:
You may create a union type SearchProps with the return type of mapstatetoprops and mapdispatchtoprops